src/sheet_metal_conus.py

```diff
@@ -3,16 +3,37 @@
 # Distributed under the terms of the GNU Lesser General Public License v3.0
 
 import sys
-#sys.path.append('/usr/share/inkscape/extensions')
 import math
 import simplestyle
 import simpletransform
 import inkex
+from copy import deepcopy
 
+# Helper functions
 def calc_angle_between_points(p1, p2):
-        xDiff = p2[0] - p1[0]
-        yDiff = p2[1] - p1[1]
-        return math.degrees(math.atan2(yDiff, xDiff))
+    xDiff = p2[0] - p1[0]
+    yDiff = p2[1] - p1[1]
+    return math.degrees(math.atan2(yDiff, xDiff))
+def calc_dist_between_points(p1, p2):
+    xDiff = p2[0] - p1[0]
+    yDiff = p2[1] - p1[1]
+    return math.sqrt(yDiff*yDiff + xDiff*xDiff)
+def normalize(p1, p2):
+    " p1,p2 defines a vector return normalized "
+    xDiff = p2[0] - p1[0]
+    yDiff = p2[1] - p1[1]
+    magn = calc_dist_between_points(p1,p2)
+    return (xDiff/magn, yDiff/magn)
+def polar_to_cartesian(cx, cy, radius, angle_degrees):
+    " So we can make arcs in the 'A' svg syntax. "
+    angle_radians = math.radians(angle_degrees)
+    return (cx + (radius * math.cos(angle_radians)),
+            cy + (radius * math.sin(angle_radians)))
+def point_on_circle(radius, angle):
+    " return xy coord of the point at distance radius from origin at angle "
+    x = radius * math.cos(angle)
+    y = radius * math.sin(angle)
+    return [x, y]
```

Comment:
all above new functions are used to calculate various aspects of arrow placement